cmd/ipfshelper/ipfshelper.go

```diff
@@ -0,0 +1,230 @@
+	return nil
+}
+
+func (h *IpfsHelper) GetPeerHostAddresses() ([]string, error) {
+	addresses, err := peer.AddrInfoToP2pAddrs(host.InfoFromHost(h.node.PeerHost))
+	if err != nil {
+		return []string{}, err
+	}
+	addressesStrings := make([]string, len(addresses))
+	for i, a := range addresses {
+		addressesStrings[i] = a.String()
+	}
+	return addressesStrings, nil
+}
+
+func (h *IpfsHelper) DownloadFile(ctx context.Context, cidString string, destinationDirectory string) (string, error) {
+	cidPath := icorepath.New(cidString)
+	resolvedPath, err := h.api.ResolvePath(ctx, cidPath)
+	if err != nil {
+		return "", fmt.Errorf("failed to resolve path: %w", err)
+	}
+	rootNodeDirectory, err := h.api.Unixfs().Get(ctx, cidPath)
+	if err != nil {
+		return "", fmt.Errorf("could not get file with CID: %w", err)
+	}
+	outputFilePath := filepath.Join(destinationDirectory, resolvedPath.Cid().String())
+	err = files.WriteTo(rootNodeDirectory, outputFilePath)
+	if err != nil {
+		return "", fmt.Errorf("could not write out the fetched CID: %w", err)
+	}
+	return outputFilePath, nil
+}
+
+func (h *IpfsHelper) AddFile(ctx context.Context, filePath string, includeHidden bool) (path.Resolved, error) {
+	fileInfo, err := os.Stat(filePath)
+	if err != nil {
+		return nil, err
+	}
+	fileNode, err := files.NewSerialFile(filePath, includeHidden, fileInfo)
+	if err != nil {
+		return nil, err
+	}
+	return h.api.Unixfs().Add(ctx, fileNode)
+}
+
+func CreateIpfsHelper(ctx context.Context, repoDirectory string, clientOnly bool) (*IpfsHelper, error) {
+	return createIpfsHelperImpl(ctx, repoDirectory, clientOnly, []string{}, "")
+}
+
+func (h *IpfsHelper) Close() error {
+	return h.node.Close()
+}
+
+func setupPlugins() error {
+	plugins, err := loader.NewPluginLoader("")
+	if err != nil {
+		return fmt.Errorf("error loading plugins: %w", err)
+	}
+	// Load preloaded and external plugins
+	if err := plugins.Initialize(); err != nil {
+		return fmt.Errorf("error initializing plugins: %w", err)
+	}
+	if err := plugins.Inject(); err != nil {
+		return fmt.Errorf("error initializing plugins: %w", err)
+	}
+	return nil
+}
+
+var loadPluginsOnce sync.Once
+
+func createIpfsHelperImpl(ctx context.Context, repoDirectory string, clientOnly bool, peerList []string, profiles string) (*IpfsHelper, error) {
+	var onceErr error
+	loadPluginsOnce.Do(func() {
+		onceErr = setupPlugins()
+	})
+	if onceErr != nil {
+		return nil, onceErr
+	}
+	client := IpfsHelper{}
+	err := client.createRepo(repoDirectory, profiles)
+	if err != nil {
+		return nil, err
+	}
+	err = client.createNode(ctx, clientOnly)
+	if err != nil {
+		return nil, err
+	}
+	err = client.connectToPeers(ctx, peerList)
+	if err != nil {
+		return nil, err
+	}
+	return &client, nil
+}
+
+func CanBeIpfsPath(pathString string) bool {
+	if len(pathString) < 5 {
+		return false
+	}
+	prefix := pathString[0:5]
+	_, err := ipfspath.ParsePath(pathString)
+	return err == nil || prefix == "/ipfs/" || prefix == "/ipld/" || prefix == "/ipns/"
```

Comment:
I've also seen an `ipfs://` prefix in the wild, could we support that?

Reply:
yes we can, and probably even should as the `init.url` config is supposed to be an URL